lib/utils.js

```diff
@@ -264,7 +265,11 @@ exports.merge = function merge(to, from, options, path) {
       continue;
     }
     if (to[key] == null) {
-      to[key] = from[key];
+      if (isPOJO(from[key])) {
+        to[key] = { ...from[key] };
+      } else {
+        to[key] = from[key];
+      }
```

Comment:
Should we also shallow copy arrays, since they're subject to the same behavior as POJOs? I can't think of a specific scenario where it's currently broken, but feels inexpensive and safer to also shallow copy arrays.
```suggestion
      if (isPOJO(from[key])) {
        to[key] = { ...from[key] };
      } else if (Array.isArray(from[key])) {
        to[key] = from[key].slice();
      } else {
        to[key] = from[key];
      }
```